docs/en/writing-running-appium/android/espresso-datamatcher-selector.md

```diff
@@ -0,0 +1,338 @@
+    }
+}, {
+    contentDescription = Rotating Button,
+    title = Rotating Button,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.RotatingButton
+    }
+}, {
+    contentDescription = ScrollBars,
+    title = ScrollBars,
+    intent = Intent {
+        cmp = io.appium.android.apis / .ApiDemos(has extras)
+    }
+}, {
+    contentDescription = Search View,
+    title = Search View,
+    intent = Intent {
+        cmp = io.appium.android.apis / .ApiDemos(has extras)
+    }
+}, {
+    contentDescription = Secure View,
+    title = Secure View,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.SecureView
+    }
+}, {
+    contentDescription = Seek Bar,
+    title = Seek Bar,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.SeekBar1
+    }
+}, {
+    contentDescription = Spinner,
+    title = Spinner,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.Spinner1
+    }
+}, {
+    contentDescription = Splitting Touches across Views,
+    title = Splitting Touches across Views,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.SplitTouchView
+    }
+}, {
+    contentDescription = Switches,
+    title = Switches,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.Switches
+    }
+}, {
+    contentDescription = System UI Visibility,
+    title = System UI Visibility,
+    intent = Intent {
+        cmp = io.appium.android.apis / .ApiDemos(has extras)
+    }
+}, {
+    contentDescription = Tabs,
+    title = Tabs,
+    intent = Intent {
+        cmp = io.appium.android.apis / .ApiDemos(has extras)
+    }
+}, {
+    contentDescription = TextClock,
+    title = TextClock,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.TextClockDemo
+    }
+}, {
+    contentDescription = TextFields,
+    title = TextFields,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.TextFields
+    }
+}, {
+    contentDescription = TextSwitcher,
+    title = TextSwitcher,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.TextSwitcher1
+    }
+}, {
+    contentDescription = Visibility,
+    title = Visibility,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.Visibility1
+    }
+}, {
+    contentDescription = WebView,
+    title = WebView,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.WebView1
+    }
+}, {
+    contentDescription = WebView2,
+    title = WebView2,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.WebView2
+    }
+}, {
+    contentDescription = WebView3,
+    title = WebView3,
+    intent = Intent {
+        cmp = io.appium.android.apis / .view.WebView3
+    }
+}
+```
+
+This lists all of the adapters that "back up" the `ListView` and can be targeted using a datamatcher selector.
+
+Here's a code snippet that shows how to locate and click `TextClock`:
+
+```js
+driver.findElementById("list")
```

Comment:
there should be a comment about which language it is

Reply:
Second for it